python/hail/dataset.py

```diff
@@ -1326,6 +1326,60 @@ def sample_variants(self, fraction, seed=1):
 
         return VariantDataset(self.hc, self._jvds.sampleVariants(fraction, seed))
 
+    @handle_py4j
+    @record_method
+    @typecheck_method(key=str)
```

Comment:
missed this before -- this should be `strlike`, because python 2 annoyingly has both `str` and `unicode`, which are different classes but behave the same in most circumstances.